### DISCUSSION TITLE
Any plans on migrating the codebase to use TypeScript?

### DISCUSSION THREAD

**Discussion by YassinEldeeb** [artilleryio/artillery#1417] [Artillery]
It's pretty hard to manage a quite large project with a loosely typed language and It's easily prone to producing bugs.

long story short: JavaScript is pretty annoying.

#### Note:
I'm willing to rewrite crucial parts of the codebase(core) to TypeScript on my own, just wanna make sure type safety is something you're interested in(I'm not gonna rewrite it manually cause I'm experimenting with an automated tool I'm developing for migrating JS projects to Typescript)

**Comment by hassy** [comment]
Not at this moment.
